source-http-ingest/src/transactor.rs

```diff
@@ -0,0 +1,144 @@
+use crate::connector_protocol::capture::{Request, Response};
+use crate::{read_capture_request, to_raw_value, write_capture_response};
+use anyhow::Context;
+use serde::Serialize;
+use tokio::{
+    io,
+    sync::{mpsc, oneshot},
+};
+
+pub struct Acknowledgements(io::BufReader<io::Stdin>);
+impl Acknowledgements {
+    pub async fn next_ack(&mut self) -> anyhow::Result<anyhow::Result<()>> {
+        let Request::Acknowledge{} = read_capture_request(&mut self.0).await? else {
+            anyhow::bail!("expected Acknowledge message");
+        };
+        Ok(Ok(()))
+    }
+}
+
+pub struct Emitter(io::Stdout);
+
+impl Emitter {
+    pub async fn emit_doc(&mut self, binding: u32, doc: &impl Serialize) -> anyhow::Result<()> {
+        use tokio::io::AsyncWriteExt;
+
+        // We intentionally don't use `write_capture_response` here because we don't want to wait
+        // for the flush call.
+        let doc = to_raw_value(doc).context("serializing output document")?;
+
+        let resp = serde_json::to_vec(&Response::Document { binding, doc })
+            .context("serializing response")?;
+        self.0.write_all(&resp).await.context("writing response")?;
+        self.0
+            .write_u8(b'\n')
+            .await
+            .context("writing response newline")?;
+        Ok(())
+    }
+
+    pub async fn commit(&mut self, cp: &impl Serialize, merge_patch: bool) -> anyhow::Result<()> {
```

Comment:
Seems like we're always using the same values for `cp` and `merge_patch` so maybe those don't need to be parameters

Reply:
This is one of the few bits of code that I was thinking would be useful for other capture connectors, which is why I thought to keep it generic. I'm kind of inclined to keep it like that, unless you feel strongly otherwise.

Reply:
Nope sounds good to me, I can see how it would be useful to possibly extract in the future as well.